Tasks/Common/webdeployment-common/jsonvariablesubstitutionutility.ts

```diff
@@ -64,6 +64,66 @@ export function substituteJsonVariable(jsonObject, envObject) {
     }
 }
 
+function stripJsonComments(content) {
+    if (!content || (content.indexOf("//") < 0 && content.indexOf("/*") < 0)) {
+        return content;
+    }
+
+    var currentChar;
+    var nextChar;
+    var prevChar;
+    var insideQuotes = false;
+    var contentWithoutComments = '';
+    var insideComment = 0;
+    var singlelineComment = 1;
+    var multilineComment = 2;
+
+    for (var i = 0; i < content.length; i++) {
+        currentChar = content[i];
+        nextChar = i + 1 < content.length ? content[i + 1] : "";
+
+        if (insideComment) {
+            var update = false;
+            if (insideComment == singlelineComment && (currentChar + nextChar === '\r\n' || currentChar === '\n')) {
+                i--;
```

Comment:
Should you move forward?

Reply:
This iterator(i) movement here just honors the new line coming after single line comment.
We should be good here, despite iterator backward movement (i--).